### PR TITLE
Perform tracer initialization in initializer

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
This actually simplifies `configure` quite a bit by ensuring that all attributes are defined in the initializer. It also means that `configure` isn't called twice.

## Checklist
- [x] ~Added to the correct milestone.~
- [x] ~Tests provided or description of manual testing performed is included in the code or PR.~ Existing tests cover this functionality.
- [x] ~Library documentation is updated.~
- [x] ~[Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).~

### CODE REVIEW

File: ddtrace/tracer.py [hunk 1]
```diff
@@ -249,52 +255,44 @@ def configure(
...
-                        port = prev_url_parsed.port
-                    scheme = "https" if https else "http"
-                    url = "%s://%s:%s" % (scheme, hostname, port)
-            elif hasattr(self, "writer") and isinstance(self.writer, AgentWriter):
-                # Reuse the URL from the previous writer if there was one.
-                url = self.writer.agent_url
+        if any(x is not None for x in [hostname, port, uds_path, https]):
+            # If any of the parts of the URL have updated, merge them with
+            # the previous writer values.
+            if hasattr(self, "writer") and isinstance(self.writer, AgentWriter):
```
Comment (P403n1x87) [inline ddtrace/tracer.py:261]: With the changes in the `__init__` method, `hasattr(self, "writer")` always evaluates to `True` now

Comment (Kyle-Verhoog) [reply ddtrace/tracer.py:261]: Good catch! I missed porting it from #2121

**Comment by Kyle-Verhoog** [comment]
There's one failing test `test_tracer_url` that fails because no validation is done on the agent url (which was previously done in the initializer). This functionality is added in #2138.

**Review by P403n1x87 — DISMISSED** [review]

**Review by jd — APPROVED** [review]

File: ddtrace/internal/agent.py [hunk 2]
```diff
@@ -35,7 +35,6 @@ def get_trace_url():
     Raises a ``ValueError`` if the URL is not supported by the Agent.
     """
     url = os.environ.get("DD_TRACE_AGENT_URL", "http://%s:%d" % (get_hostname(), get_trace_port()))
-    verify_url(url)
```
Comment (Kyle-Verhoog) [inline ddtrace/internal/agent.py:38]: This is no longer necessary as we can verify the URL before instantiating an AgentWriter

**Review by P403n1x87 — APPROVED** [review]

**Review by majorgreys — APPROVED** [review]
👍🏽

**Pull request merged**